### ISSUE TITLE
✨ Backend for email verification

### ISSUE THREAD

**Issue by gobyrne** [GCTC-NTGC/gc-digital-talent#10374]
## ✨ Feature

The email verification feature requires a few new things on the backend to store verifications and similar.

## 🕵️ Details

> [!NOTE] 
Addition of work email was spun off to #10574 

## 🙋‍♀️ Proposed Implementation

https://github.com/GCTC-NTGC/gc-digital-talent/pull/10505

## ✅ Acceptance Criteria

- [ ] user fields to store timestamps when email was verified (default null)
- [ ] API mutation to request verification email with a code to either email
  - [ ] Requesting a code should invalidate a previous code
  - [ ] Changing the email address should invalidate an unredeemed code
  - [ ] Changing the email should clear existing email verification
  - [ ] codes should be valid for 2 hours
  - [ ] code complexity should be 6 alphanumeric chars (all caps, but verification is case insensitive and all whitespace is removed before checking)
  - [ ] email should use provisional templates in GCNotify
- [ ] mutation to redeem a code to mark either email as verified
  - [ ] must be logged in and user matches user who requested code
  - [ ] stored email should still match email attached to code
- [ ] automated tests

Out of scope: any UI except for the email itself

## 🛑 Blockers

```[tasklist]
### Blocked By
```
